src/pages/api/v1/applications/[id].ts

```diff
@@ -7,15 +7,16 @@ const getApplication = async (id: number, res: NextApiResponse) => {
     const data = await prisma.application.findUnique({
       where: { id },
     });
-    if (data) {
-      res.status(200).json(data);
-    } else {
-      res
+
+    if (!data) {
+      return res
         .status(404)
         .json({ message: `Could not find Application with id ${id}` });
     }
+
+    return res.status(200).json(data);
```

Comment:
My personal preference for me is to always have the success response last.

If we have any error responses, I like to early exit and `return` the response.

If you don't return the response, it will cascade. So the last one will override the previous responses. (You'll find this behaviour in Express as well.)